File: TodayExtension/TodayExtension-Today/TodayViewController.m
//
//  TodayViewController.m
//  TodayExtension-Today
//
//  Created by 김기범 on 2016. 7. 18..
//  Copyright © 2016년 김기범. All rights reserved.
//

#import "TodayViewController.h"
#import <NotificationCenter/NotificationCenter.h>

@interface TodayViewController () <NCWidgetProviding>

@end

@implementation TodayViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(refresh)
                                                 name:NSUserDefaultsDidChangeNotification
                                               object:nil];
    [self refresh];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)refresh {
    NSUserDefaults *sharedUserDefaults = [[NSUserDefaults alloc] initWithSuiteName:@"group.org.nhnnext.kiboom"];
    self.label.text = [sharedUserDefaults objectForKey:@"Text"];
}

@end

File: Week 6-2 (Extension)/Today/TodayViewController.swift
//
//  TodayViewController.swift
//  Today
//
//  Created by 김기범 on 2016. 7. 20..
//  Copyright © 2016년 김기범. All rights reserved.
//

import UIKit
import NotificationCenter

class TodayViewController: UIViewController, NCWidgetProviding {
        
    var userDefaults: NSUserDefaults = NSUserDefaults.init(suiteName: "group.org.nhnnext.kiboom")!
    @IBOutlet var dDayLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func viewDidAppear(animated: Bool) {
        update()
    }
    
    override func viewWillDisappear(animated: Bool) {
        print("뷰 윌 디써피어")
    }
    
    override func viewDidDisappear(animated: Bool) {
        print("뷰 디드 디써피어")
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    func update() {
        guard let dDay = userDefaults.stringForKey("dDay") else {
            dDayLabel.text = "앱에서 날짜를 설정하세요"
            return
        }
        dDayLabel.text = dDay
    }
    
    func widgetMarginInsetsForProposedMarginInsets(defaultMarginInsets: UIEdgeInsets) -> UIEdgeInsets {
        return UIEdgeInsetsZero
    }
    
    func widgetPerformUpdateWithCompletionHandler(completionHandler: ((NCUpdateResult) -> Void)) {
        // Perform any setup necessary in order to update the view.

        // If an error is encountered, use NCUpdateResult.Failed
        // If there's no update required, use NCUpdateResult.NoData
        // If there's an update, use NCUpdateResult.NewData
        completionHandler(NCUpdateResult.NewData)
    }
    
}
